gui/src/app/init/analytics.js

```diff
@@ -1,75 +1,88 @@
-/* Copyright (C) 2017 Canonical Ltd. */
+/* Copyright (C) 2019 Canonical Ltd. */
 'use strict';
 
-
-/**
-  Factory that returns a function for sending analytics to GA.
-
-  @param {String} controllerAPI For checking status of the user.
-  @param {String} dataLayer The GA dataLayer instance.
-  @return {function} The curryed function for sending analaytics.
-*/
-const sendAnalyticsFactory = function(controllerAPI, dataLayer) {
+class Analytics {
+  /**
+    Set up the analytics instance.
+    @param {String} dataLayer - The Google Tag Manager dataLayer instance.
+    @param {Object} data - Optional parameters:
+      - categories: an array of categories;
+      - getLabel: a function that will be used to generate the label string.
+  */
+  constructor(dataLayer, data={}) {
+    this.dataLayer = dataLayer;
+    this.categories = data.categories || [];
+    this.getLabel = data.getLabel;
+    // Common events.
+    this.CANCEL = 'Cancel';
+    this.CLICK = 'Click';
+    this.UPDATE = 'Update';
+    this.VIEW = 'View';
+  }
 
   /**
-    The retunrned function that sends stats to GA.
+    Append a category. This will be displayed in the analytics as
+    something like: Charmbrowser > Search Results > Filter
+    @param {Object|String} categoryOrComponent - A category name or React
+      component.
+    @returns {Object} A new Analytics instance including the old and new data.
+  */
+  addCategory(categoryOrComponent) {
+    let name;
+    if (typeof categoryOrComponent === 'string') {
+      name = categoryOrComponent;
```

Comment:
We should validate that it doesn't contain the string ">" as that could throw off parsing of event trees.

Reply:
Do you think we should replace it with something else?